### PR TITLE
chore(eoa): refactor accounts

### PR DESCRIPTION
**Pull request by Lazar955**
moves accounts in central place in EOA

task: none

### CODE REVIEW

**Review by idea404 — APPROVED** [review]

File: e2e/app/eoa/static.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		},
+	},
+	netconf.Testnet: {
+		{
+			Type:          TypeRemote,
+			Role:          RoleCreate3Deployer,
+			Address:       testnetCreate3Deployer,
+			Chains:        ChainSelectorAll,
+			MinBalance:    big.NewInt(2),
+			TargetBalance: big.NewInt(10),
```
Comment (kevinhalliday) [inline e2e/app/eoa/static.go:123]: I don't think create3 deployers need this much - they only ever do one deployment

**Review by kevinhalliday — APPROVED** [review]

File: e2e/app/eoa/eoa.go [hunk 2]
```diff
@@ -4,85 +4,128 @@ package eoa
...
-
-	addr, ok := secureAddrs[network][typ]
+// PrivateKey returns the private key for the EOA identified by the network and role.
+func PrivateKey(ctx context.Context, network netconf.ID, role Role) (*ecdsa.PrivateKey, error) {
+	acc, ok := accountForRole(network, role)
 	if !ok {
-		return nil, errors.New("eoa key not defined", "network", network, "type", typ)
+		return nil, errors.New("eoa key not defined", "network", network, "role", role)
+	}
+	if network == netconf.Devnet {
```
Comment (corverroos) [inline e2e/app/eoa/eoa.go:88]: this should be based on type:
- if type == wellknown, return privkey
- if type == remote, return error
- if type == secret, download

File: e2e/app/eoa/eoa.go [hunk 3]
```diff
@@ -4,85 +4,128 @@ package eoa
...
+	TypeRemote    Type = "remote"     // stored in (fireblocks) accessible via API to sign
+	TypeSecret    Type = "secret"     // stored in GCP can be downloaded to disk
+	TypeWellKnown Type = "well-known" // well-known eoa private keys in the repo
 )
 
-func (t Type) Verify() error {
-	if t != TypeRelayer && t != TypeMonitor {
-		return errors.New("invalid type", "type", t)
-	}
+type ChainSelector func(netconf.Network) []netconf.Chain
```
Comment (corverroos) [inline e2e/app/eoa/eoa.go:35]: this type and impls below can be unexported

File: e2e/app/eoa/static.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+//nolint:gochecknoglobals // Static mappings.
+var statics = map[netconf.ID][]Account{
+	netconf.Devnet: {
+		{
+			Type:          TypeRemote,
+			Role:          RoleCreate3Deployer,
+			Address:       devnetCreate3Deployer,
+			Chains:        ChainSelectorAll,
+			MinBalance:    big.NewInt(10),
```
Comment (corverroos) [inline e2e/app/eoa/static.go:20]: balancers are in wei, so this is 10 wei, which is basically 0. I suggest you create some globals:
```
var (
  ether1 = new(big.Int).Mul(big.NewInt(1),big.NewInt(params.Ether)
  ether5 = new(big.Int).Mul(big.NewInt(5),big.NewInt(params.Ether) 
  ether10 ...
  etc
)
```

File: e2e/app/eoa/static.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+			Role:          RoleAdmin,
+			Address:       devnetAdmin,
+			Chains:        ChainSelectorAll,
+			MinBalance:    big.NewInt(10),
+			TargetBalance: big.NewInt(100),
+		},
+		{
+			Type:          TypeWellKnown,
+			Role:          RoleRelayer,
+			Address:       crypto.PubkeyToAddress((anvil.DevPrivateKey5()).PublicKey),
```
Comment (corverroos) [inline e2e/app/eoa/static.go:42]: anvil package has address vars, no need to convert

File: e2e/app/eoa/eoa.go [hunk 6]
```diff
@@ -4,85 +4,128 @@ package eoa
...
+			if account.Role == role {
+				addresses = append(addresses, account.Address)
+			}
+		}
+	}
+
+	return addresses
+}
+
+func addr(hex string) common.Address {
```
Comment (corverroos) [inline e2e/app/eoa/eoa.go:129]: rather use common.HexToAddress(hex) directly, this abbreviation isn't super clear

File: e2e/app/eoa/static.go (hunk 4 above)
Comment (corverroos) [reply e2e/app/eoa/static.go:20]: Suggest also to group threshold balances into 3 groups maybe:
```
var (
  minBalanceSmall = gwei100
  targetBalanceSmall = gwei500
  
  minBalanceMedium = ether1
  targetBalanceMedium = ether5
  
  minBalanceLarge = ether10
  targetBalanceLarge = ether100 
)
```

then map accounts to one of these. Please confirm these balances in slack thread with the team

File: e2e/cmd/cmd.go [hunk 7]
```diff
@@ -223,12 +223,12 @@ func verifyKeyNodeType(def app.Definition, cfg key.UploadConfig) error {
 	}
 
 	if cfg.Type == key.EOA {
-		eoaType := eoa.Type(cfg.Name)
-		if err := eoaType.Verify(); err != nil {
+		eoaRole := eoa.Role(cfg.Name)
```
Comment (corverroos) [inline e2e/cmd/cmd.go:226]: we should very type==secret and role==valid

File: lib/contracts/portal/deploy.go [hunk 8]
```diff
@@ -86,69 +87,69 @@ func getDeployCfg(chainID uint64, network netconf.ID, valSetID uint64, vals []bi
 
 func mainnetCfg() DeploymentConfig {
 	return DeploymentConfig{
-		Create3Factory: contracts.MainnetCreate3Factory(),
-		Create3Salt:    contracts.PortalSalt(netconf.Mainnet),
-		Owner:          contracts.MainnetPortalAdmin(),
-		Deployer:       contracts.MainnetDeployer(),
+		Create3Factory: eoa.MainnetCreate3Factory(),
+		Create3Salt:    eoa.PortalSalt(netconf.Mainnet),
```
Comment (corverroos) [inline lib/contracts/portal/deploy.go:91]: PortalSalt is not a eoa thing, it is a contract thing, move it back pls.

File: lib/contracts/portal/deploy.go [hunk 9]
```diff
@@ -86,69 +87,69 @@ func getDeployCfg(chainID uint64, network netconf.ID, valSetID uint64, vals []bi
...
-		ExpectedAddr:   contracts.DevnetPortal(),
+		ExpectedAddr:   eoa.DevnetPortal(),
 	}
 }
 
 func AddrForNetwork(network netconf.ID) (common.Address, bool) {
 	switch network {
 	case netconf.Mainnet:
-		return contracts.MainnetPortal(), true
+		return eoa.MainnetPortal(), true
```
Comment (corverroos) [inline lib/contracts/portal/deploy.go:146]: portals are not eoa, they are contracts, move it back pls

File: lib/contracts/portal/deploy.go [hunk 10]
```diff
@@ -86,69 +87,69 @@ func getDeployCfg(chainID uint64, network netconf.ID, valSetID uint64, vals []bi
...
 }
 
 func devnetCfg(valSetID uint64, vals []bindings.Validator) DeploymentConfig {
 	return DeploymentConfig{
-		Create3Factory: contracts.DevnetCreate3Factory(),
-		Create3Salt:    contracts.PortalSalt(netconf.Devnet),
-		Owner:          contracts.DevnetPortalAdmin(),
-		Deployer:       contracts.DevnetDeployer(),
-		ProxyAdmin:     contracts.DevnetProxyAdmin(),
+		Create3Factory: eoa.DevnetCreate3Factory(),
```
Comment (corverroos) [inline lib/contracts/portal/deploy.go:130]: create3factory is a contract, not a eoa, move it back pls

File: lib/contracts/proxyadmin/deploy.go [hunk 11]
```diff
@@ -60,54 +61,54 @@ func getDeployCfg(chainID uint64, network netconf.ID) (DeploymentConfig, error)
...
-		Create3Factory: contracts.DevnetCreate3Factory(),
-		Create3Salt:    contracts.ProxyAdminSalt(netconf.Devnet),
-		Owner:          contracts.DevnetProxyAdminOwner(),
-		Deployer:       contracts.DevnetDeployer(),
-		ExpectedAddr:   contracts.DevnetProxyAdmin(),
+		Create3Factory: eoa.DevnetCreate3Factory(),
+		Create3Salt:    eoa.ProxyAdminSalt(netconf.Devnet),
+		Owner:          eoa.DevnetProxyAdminOwner(),
+		Deployer:       eoa.DevnetDeployer(),
+		ExpectedAddr:   eoa.DevnetProxyAdmin(),
```
Comment (corverroos) [inline lib/contracts/proxyadmin/deploy.go:98]: DevnetProxyAdmin is a contract, not a eoa

**Review by corverroos — CHANGES_REQUESTED** [review]

File: e2e/app/fund.go [hunk 12]
```diff
@@ -38,24 +37,24 @@ func accountsToFund(network netconf.ID) []common.Address {
...
-			eoa.MustAddress(netconf.Staging, eoa.TypeRelayer),
-			eoa.MustAddress(netconf.Staging, eoa.TypeMonitor),
+			eoa.StagingCreate3Deployer(),
+			eoa.StagingDeployer(),
+			eoa.StagingProxyAdminOwner(),
+			eoa.StagingPortalAdmin(),
+			eoa.StagingAVSAdmin(),
+			eoa.MustAddress(netconf.Staging, eoa.RoleRelayer),
+			eoa.MustAddress(netconf.Staging, eoa.RoleMonitor),
 		}
```
Comment (corverroos) [inline e2e/app/fund.go:47]: you currently have two ways to access an eoa address, lets align on only a single way, I suggest: `eoa.MustAddress(netconf.Staging, eoa.RoleRelayer)`

File: e2e/cmd/cmd.go (hunk 7 above)
Comment (corverroos) [reply e2e/cmd/cmd.go:226]: you still don't check that the eoa type==secret

File: e2e/app/fund.go [hunk 13]
```diff
@@ -37,25 +30,25 @@ func accountsToFund(network netconf.ID) []common.Address {
...
 		return []common.Address{
-			fbDev,
-			contracts.StagingCreate3Deployer(),
-			contracts.StagingDeployer(),
-			contracts.StagingProxyAdminOwner(),
-			contracts.StagingPortalAdmin(),
-			contracts.StagingAVSAdmin(),
-			eoa.MustAddress(netconf.Staging, eoa.TypeRelayer),
-			eoa.MustAddress(netconf.Staging, eoa.TypeMonitor),
+			eoa.MustAddress(netconf.Staging, eoa.RoleFbDev),
```
Comment (corverroos) [inline e2e/app/fund.go:33]: lets reduce some duplication:
```
return eoa.MustAddresses(netconf.Staging, eao.AllRoles()...)
```

**Review by corverroos — APPROVED** [review]

**Pull request merged**